events/pkg/kv/kv.go

```diff
@@ -0,0 +1,75 @@
+package kv
+
+import (
+	"errors"
+	"time"
+
+	"github.com/nats-io/nats.go"
+)
+
+var (
+	mgr nats.KeyValueManager
+)
+
+var (
+	ErrKVUninitialized = errors.New("KV subsystem is uninitialized")
+)
+
+// InitializeKV must be called before anything else. It's safe to call multiple times.
+func InitializeKV(js nats.JetStreamContext) {
+	if mgr == nil {
+		mgr = js
+	}
+}
+
+// DefaultKVConfig returns a configuration with "mostly sane" defaults. Override
+// with the following Option functions
+func DefaultKVConfig(bucketName string) *nats.KeyValueConfig {
```

Comment:
Also update [NatsOptions](https://github.com/metal-toolbox/hollow-toolbox/blob/2a97d4938fe3630f5684515c74b9e9455938a2ac/events/nats_config.go#L34) with a struct for the required KV store parameters, and have the KV store conditionally initialized through [Open()](https://github.com/metal-toolbox/hollow-toolbox/blob/2a97d4938fe3630f5684515c74b9e9455938a2ac/events/nats.go#L67)

Reply:
Do you have a particular use-case in mind for this? My thinking is that `ControllerRegistry` (and eventually `ConditionState` and `ConditionHistory`) have specific requirements, those options would be coded specifically in their libraries to avoid configuration mistakes. We _could_ add this to `NatsOptions` if you see a use for it, though.